instrumentation/opentelemetry-instrumentation-sqlalchemy/src/opentelemetry/instrumentation/sqlalchemy/engine.py

```diff
@@ -300,3 +298,22 @@ def _get_attributes_from_cursor(vendor, cursor, attrs):
             if info.port:
                 attrs[SpanAttributes.NET_PEER_PORT] = int(info.port)
     return attrs
+
+
+def _get_connection_string(engine):
+    drivername = engine.url.drivername or ""
+    host = engine.url.host or ""
+    port = engine.url.port or ""
+    database = engine.url.database or ""
+    return f"{drivername}://{host}:{port}/{database}"
+
+
+def _get_attributes_from_engine(engine):
+    """Set metadata attributes of the database engine"""
+    attrs = {}
+
+    attrs["pool.name"] = getattr(
```

Comment:
You save the pool_name/connection_string just for __init__, 
Is there a possibility that one of these fields will change after it?

Reply:
This is a good point. Based on my inpection of SQLAlchemy code, `logging_name` should not change unless user is overriding private attributes. However, `engine.url` is not considered as read-only and this seems like a valid case

```
from sqlalchemy import create_engine
from sqlalchemy.engine import make_url
engine = create_engine("sqlite:///test.db")
engine.url = make_url("sqlite:///test2.db")
```
which makes me think we might still have to rely on weak references...